File: main/Visualization.cpp
#include <Arduino.h>
#include "Adafruit_NeoPixel.h"

#define STRIP_FRONT_LEFT 6
#define STRIP_RIGHT_TOP 7

#define LED_COUNT 150 // leds per matrix

#define X 15
#define Y 15

#define FPS 10


enum class Side { FRONT, LEFT, TOP, RIGHT };

#define BLANK Adafruit_NeoPixel::Color(0, 0, 0)
#define GREEN Adafruit_NeoPixel::Color(0, 255, 0)
#define BLUE Adafruit_NeoPixel::Color(0, 0, 255)
#define RED Adafruit_NeoPixel::Color(255, 0, 0)

class Visualization
{
public:

  Visualization()
  {
    front_left.begin();
    right_top.begin();

    front_left.setBrightness(255);
    right_top.setBrightness(255);
  }

  /**
    draws a pixel on the 4-sided cube
  **/
  int drawPixel(Side side, int x, int y, uint32_t color)
  {
    int i; // strip index

    if (side == Side::FRONT) 
    {
        if(y % 2 == 0) // ->
        {
          i = x + (y * X) + (X * (y + 1));
        }
        else // <-
        {
          i = (y * X) + (X * (y + 1)) - 1 - x;
        }
    }
    else if (side == Side::LEFT) 
    {
        if(y % 2 == 0) // ->
        {
          i = x + (y * X * 2);
        }
        else          // <-
        {
          i = (X - x - 1) + (y * (X * 2) + X);
        }
    }
    else if (side == Side::RIGHT) 
    {
      // right_top
    }
    else if (side == Side::TOP) 
    {
      // right_top
    }
    Serial.println(i);
    return -1;
  }

private:

  Adafruit_NeoPixel front_left = Adafruit_NeoPixel(LED_COUNT, STRIP_FRONT_LEFT, NEO_GRB + NEO_KHZ800);
  Adafruit_NeoPixel right_top = Adafruit_NeoPixel(LED_COUNT, STRIP_RIGHT_TOP, NEO_GRB + NEO_KHZ800);
};

File: Visualization.py
from neopixel import NeoPixel
from machine import Pin, Timer
#from enum import Enum

class Side:
    FRONT = 1
    LEFT = 2
    TOP = 3
    RIGHT = 4


class Visualization:

    MATRIX_SIZE = 15

    def __init__(self):
    
        self.pixels_left_front = NeoPixel(Pin(29, Pin.OUT), self.MATRIX_SIZE * self.MATRIX_SIZE * 2)
        self.pixels_top_right = NeoPixel(Pin(28, Pin.OUT), self.MATRIX_SIZE * self.MATRIX_SIZE * 2)

    def draw_pixel(self, side, x, y, color):
        # todo remove
        self.pixels_left_front.fill((0,0,0))
        self.pixels_top_right.fill((0,0,0))

        if x >= self.MATRIX_SIZE or x < 0:
            return False
        if y >= self.MATRIX_SIZE or y < 0:
            return False

        i = 0  # strip index
        max_val = self.MATRIX_SIZE * self.MATRIX_SIZE * 2

        # --- first Strip
        if side == Side.FRONT:
            if y % 2 == 0:  # ->
                i = x + (y * self.MATRIX_SIZE) + (self.MATRIX_SIZE * (y + 1))
            else:  # <-
                i = (y * self.MATRIX_SIZE) + (self.MATRIX_SIZE * (y + 1)) - 1 - x
        elif side == Side.LEFT:
            if y % 2 == 0:  # ->
                i = x + (y * self.MATRIX_SIZE * 2)
            else:  # <-
                i = (self.MATRIX_SIZE - x - 1) + (y * (self.MATRIX_SIZE * 2) + X)

        # --- second Strip
        elif side == Side.TOP:
            if y % 2 == 0:  # <-
                i = max_val - (x + (self.MATRIX_SIZE * y * 2) + 1)
            else:  # ->
                i = max_val - ((self.MATRIX_SIZE - x - 1) + (y * (self.MATRIX_SIZE * 2) + self.MATRIX_SIZE) + 1)

        elif side == Side.RIGHT:
            # TODO: Handle RIGHT side
            pass

        if side == Side.LEFT or side == Side.FRONT:
            self.pixels_left_front[i] = color
            # todo remove
            self.pixels_left_front.write()
            self.pixels_top_right.write()
            return True
        if side == Side.TOP or side == Side.RIGHT:
            self.pixels_top_right[i] = color
            # todo remove
            self.pixels_left_front.write()
            self.pixels_top_right.write()
            return True
        print(i)
        return False


